core/API/DataTableManipulator/ReportTotalsCalculator.php

```diff
@@ -53,66 +78,32 @@ public function calculate($table)
      */
     protected function manipulateDataTable($dataTable)
     {
-        $report = $this->findCurrentReport();
-
-        if (!empty($report) && !$report->getDimension() && !$this->isReportAllMetricsReport($report)) {
+        if (!empty($this->report) && !$this->report->getDimension() && !$this->isAllMetricsReport()) {
             // we currently do not calculate the total value for reports having no dimension
             return $dataTable;
         }
 
-        // Array [readableMetric] => [summed value]
-        $totalValues = array();
-
+        $this->totals       = array();
         $firstLevelTable    = $this->makeSureToWorkOnFirstLevelDataTable($dataTable);
         $metricsToCalculate = Metrics::getMetricIdsToProcessReportTotal();
 
-        $realMetricNames = array();
+        $metricNames = array();
         foreach ($metricsToCalculate as $metricId) {
-            $metricName = Metrics::getReadableColumnName($metricId);
-            $realMetricName = $this->hasDataTableMetric($firstLevelTable, $metricId, $metricName);
```

Comment:
The report total calculation was buggy and did not calculate the total values correctly! From the beginning to find the correct metricId or metricName we always only had a look in the first row. But sometimes the first row does not have this column. In this case we did sometimes not sum some metrics at all and sometimes some values were simply ignored. We do always have to iterate over all possible metrics and over all rows to do calculate the values correctly even though it is slower.